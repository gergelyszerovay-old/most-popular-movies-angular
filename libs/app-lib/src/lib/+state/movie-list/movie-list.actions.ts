import {createAction, props} from '@ngrx/store';
import {MovieListEntity} from "../../services/tmdb.resource.models";

export const loadMovieList = createAction(
  '[MovieList] Load MovieList',
  props<{ page: number }>()
);

export interface LoadMovieListSuccessProps {
  movieList: MovieListEntity[],
  totalMovies: number,
  totalPages: number
}

export const loadMovieListSuccess = createAction(
  '[MovieList] Load MovieList Success',
  props<LoadMovieListSuccessProps>()
);

export const loadMovieListFailure = createAction(
  '[MovieList] Load MovieList Failure',
  props<{ errorMessage: string }>()
);

export const selectMovie = createAction(
  '[MovieList] Movie Select',
  props<{ id: number }>()
);
