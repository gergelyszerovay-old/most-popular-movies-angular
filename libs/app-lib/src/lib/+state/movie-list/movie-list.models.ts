import {MovieGenresEntity, MovieListEntity} from "../../services/tmdb.resource.models";

export interface MovieListEntityWithGenreNames extends MovieListEntity {
  _genres?: Array<MovieGenresEntity>;
}

