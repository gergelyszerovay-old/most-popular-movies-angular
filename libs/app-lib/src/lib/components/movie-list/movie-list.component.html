<div *ngIf="errorMessage" class="c-movie-list x-load-error">
  <ngb-toast class="bg-danger text-light">
    Error: Cannot load the movie list from the server!<br>
    <a (click)="onReloadList.emit(); $event?.preventDefault();" href="#">Click here to retry.</a>
  </ngb-toast>
</div>
<div *ngIf="!!movieList" class="c-movie-list x-no-load-error">
  <ul [class.x-is-loading]="isLoading" data-qa="movie-list-ul">
    <li *ngFor="let movie of movieList; trackBy: trackByFnMovieList">
      <a (click)="onMovieClick.emit(movie.id); $event?.preventDefault();"
         [class.x-selected]="movie.id === selectedMovie?.id" href="#">
        <span>{{movie?.title}} ({{movie?.release_date.substr(0, 4)}})</span>
        <svg class="bi bi-caret-right-fill" fill="currentColor" height="1em" viewBox="0 0 16 16" width="1em"
             xmlns="http://www.w3.org/2000/svg">
          <path
            d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"/>
        </svg>
      </a>
    </li>
  </ul>
  <ngb-pagination
    (pageChange)="onPageChange.emit($event)"
    *ngIf="paginationCollectionSize && movieList.length"
    [collectionSize]="paginationCollectionSize"
    [maxSize]="1"
    [pageSize]="pageSize"
    [page]="page"
    [rotate]="true"
  ></ngb-pagination>
</div>
