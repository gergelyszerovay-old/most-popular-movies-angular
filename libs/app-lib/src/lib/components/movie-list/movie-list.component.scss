$componentClass: '.c-movie-list';

#{$componentClass} {
  &.x-load-error {
    margin: 1rem;

    a {
      color: #fff;
      font-weight: bold;
      text-decoration: underline;
    }
  }

  &.x-no-load-error {
    ul {
      margin: 0;
      padding: 0 0 1rem;

      &.x-is-loading {
        li {
          a {
            color: #aaa;

          }
        }
      }

      li {
        list-style-type: none;
        cursor: pointer;
        height: 2rem;
        line-height: 2rem;
        border-bottom: solid #dee2e6 1px;
        padding: 0;

        a {
          position: relative;
          display: block;
          width: 100%;
          height: 100%;
          padding-left: 0.5rem;
          padding-right: 0.5rem;

          white-space: nowrap;
          overflow: hidden;
          text-overflow: ellipsis;

          color: #212529;

          svg {
            position: absolute;
            right: 0;
            top: 0.5rem;
            display: none;
            color: #212529;
          }

          &:hover {
            background-color: #eee;
            text-decoration: none;

            svg {
              display: block;
            }
          }
        }
      }
    }

    ngb-pagination {
      margin: 1rem;
      display: block;
    }
  }

  // &.x-no-load-error
}

@media (min-width: 768px) {
  #{$componentClass} {
    &.x-no-load-error {
      ul {
        li {
          a {
            &.x-selected {
              background-color: #007bff;
              color: #fff;
              text-decoration: none;

              svg {
                display: block;
                color: #fff;
              }
            }
          }
        }
      }
    }
  }
}


