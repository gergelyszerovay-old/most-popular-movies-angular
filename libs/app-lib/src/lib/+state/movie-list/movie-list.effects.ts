import {Injectable} from '@angular/core';
import {Actions, createEffect, ofType} from '@ngrx/effects';
import {map} from "rxjs/operators";
import {fetch} from '@nrwl/angular';

import * as MovieListActions from './movie-list.actions';
import {TmdbResourceService} from "../../services/tmdb.resource.service";
import {HttpErrorResponse} from "@angular/common/http";
import {MovieListRequestResult} from "../../services/tmdb.resource.models";

@Injectable()
export class MovieListEffects {
  loadMovieList$ = createEffect(() =>
    this.actions$.pipe(
      ofType(MovieListActions.loadMovieList),
      fetch({
        run: (action) => {
          return this.movieListService.getMovieList(action.page).pipe(
            map((response: MovieListRequestResult) => MovieListActions.loadMovieListSuccess({
              movieList: response.results,
              totalMovies: response.total_results,
              totalPages: response.total_pages
            }))
          );
        },

        onError: (action, error: HttpErrorResponse) => {
          return MovieListActions.loadMovieListFailure({errorMessage: error.message});
        },
      })
    )
  );

  constructor(private actions$: Actions, private movieListService: TmdbResourceService) {
  }
}
