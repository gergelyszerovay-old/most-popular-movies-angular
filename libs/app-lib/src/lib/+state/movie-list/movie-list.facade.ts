import {Injectable} from '@angular/core';

import {select, Store} from '@ngrx/store';

import * as MovieListSelectors from './movie-list.selectors';
import * as MovieListActions from './movie-list.actions';
import {loadMovieList} from './movie-list.actions';
import {AppLibRootStatePartial} from "../app-lib.reducer";

@Injectable()
export class MovieListFacade {
  allMovieList$ = this.store.pipe(select(MovieListSelectors.getAllMovieList));
  totalMovies$ = this.store.pipe(select(MovieListSelectors.getTotalMovies));
  totalPages$ = this.store.pipe(select(MovieListSelectors.getTotalPages));
  page$ = this.store.pipe(select(MovieListSelectors.getPage));
  selectedMovie$ = this.store.pipe(select(MovieListSelectors.getSelectedMovie));
  errorMessage$ = this.store.pipe(select(MovieListSelectors.getErrorMessage));
  loading$ = this.store.pipe(select(MovieListSelectors.getLoading));

  // allMovieListWithGenreNames$ = this.store.pipe(select(MovieListSelectors.getAllMovieListWithGenreNames));

  selectedMovieWithGenreNames$ = this.store.pipe(select(MovieListSelectors.getSelectedMovieWithGenreNames));

  constructor(private store: Store<AppLibRootStatePartial>) {
  }

  loadMovieList(page: number) {
    this.store.dispatch(loadMovieList({page}));
  }

  selectMovie(id: number) {
    this.store.dispatch(MovieListActions.selectMovie({id}));
  }
}
