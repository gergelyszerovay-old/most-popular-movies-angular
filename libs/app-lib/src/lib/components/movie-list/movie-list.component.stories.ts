import {NgbModule} from "@ng-bootstrap/ng-bootstrap";

import {boolean, number, object, text} from '@storybook/addon-knobs';

import {action} from '@storybook/addon-actions';

import {MovieListComponent} from './movie-list.component';

export default {
  title: 'MovieListComponent'
}

const moduleMetadata = {
  imports: [NgbModule],
}

const movieListData = [{
  "popularity": 246.384,
  "vote_count": 676,
  "video": false,
  "poster_path": "/bOKjzWDxiDkgEQznhzP4kdeAHNI.jpg",
  "id": 605116,
  "adult": false,
  "backdrop_path": "/qVygtf2vU15L2yKS4Ke44U4oMdD.jpg",
  "original_language": "en",
  "original_title": "Project Power",
  "genre_ids": [
    28,
    80,
    878
  ],
  "title": "Project Power",
  "vote_average": 6.7,
  "overview": "An ex-soldier, a teen and a cop collide in New Orleans as they hunt for the source behind a dangerous new pill that grants users temporary superpowers.",
  "release_date": "2020-08-14"
},
  {
    "popularity": 171.505,
    "vote_count": 15,
    "video": false,
    "poster_path": "/k68nPLbIST6NP96JmTxmZijEvCA.jpg",
    "id": 577922,
    "adult": false,
    "backdrop_path": "/wzJRB4MKi3yK138bJyuL9nx47y6.jpg",
    "original_language": "en",
    "original_title": "Tenet",
    "genre_ids": [
      28,
      53
    ],
    "title": "Tenet",
    "vote_average": 6.2,
    "overview": "Armed with only one word - Tenet - and fighting for the survival of the entire world, the Protagonist journeys through a twilight world of international espionage on a mission that will unfold in something beyond real time.",
    "release_date": "2020-08-22"
  },
  {
    "popularity": 139.38,
    "vote_count": 42,
    "video": false,
    "poster_path": "/eAUzmhP54bE1vPXaY7FbuZREJlR.jpg",
    "id": 594718,
    "adult": false,
    "backdrop_path": "/mBbA77FyzhU0Tz9tmbKG8heGmh3.jpg",
    "original_language": "ru",
    "original_title": "Спутник",
    "genre_ids": [
      18,
      27,
      878
    ],
    "title": "Sputnik",
    "vote_average": 6.4,
    "overview": "At the height of the Cold War, a Soviet spacecraft crash lands after a mission gone awry, leaving the commander as its only survivor. After a renowned Russian psychologist is brought in to evaluate the commander’s mental state, it becomes clear that something dangerous may have come back to Earth with him…",
    "release_date": "2020-07-14"
  }];

export const isLoaded = () => ({
  moduleMetadata,
  component: MovieListComponent,
  props: {
    onMovieClick: action('onMovieClick'),
    onPageChange: action('onPageChange'),
    onReloadList: action('onReloadList'),

    errorMessage: text('errorMessage', ''),
    isLoading: boolean('isLoading', false),

    page: number('page', 1),
    pageSize: number('pageSize', 5),
    paginationCollectionSize: number('paginationCollectionSize', 313),

    movieList: object('movieList', movieListData)
  }
})

export const isLoadedWithSelectedMovie = () => ({
  moduleMetadata,
  component: MovieListComponent,
  props: {
    onMovieClick: action('onMovieClick'),
    onPageChange: action('onPageChange'),
    onReloadList: action('onReloadList'),

    errorMessage: text('errorMessage', ''),
    isLoading: boolean('isLoading', false),

    page: number('page', 1),
    pageSize: number('pageSize', 5),
    paginationCollectionSize: number('paginationCollectionSize', 313),

    movieList: object('movieList', movieListData),
    selectedMovie: object('selectedMovie', movieListData[1])
  }
})

export const isLoading = () => ({
  moduleMetadata,
  component: MovieListComponent,
  props: {
    onMovieClick: action('onMovieClick'),
    onPageChange: action('onPageChange'),
    onReloadList: action('onReloadList'),

    errorMessage: text('errorMessage', ''),
    isLoading: boolean('isLoading', true),

    page: number('page', 1),
    pageSize: number('pageSize', 5),
    paginationCollectionSize: number('paginationCollectionSize', 313),

    movieList: object('movieList', movieListData)
  }
})

export const isError = () => ({
  moduleMetadata,
  component: MovieListComponent,
  props: {
    onMovieClick: action('onMovieClick'),
    onPageChange: action('onPageChange'),
    onReloadList: action('onReloadList'),

    errorMessage: text('errorMessage', 'Error'),
    isLoading: boolean('isLoading', false),

  }
})
