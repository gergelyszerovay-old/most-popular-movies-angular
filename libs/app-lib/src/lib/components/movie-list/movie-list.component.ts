import {ChangeDetectionStrategy, Component, EventEmitter, Input, Output} from '@angular/core';
import {MovieListEntity} from "../../services/tmdb.resource.models";
import {MovieListEntityWithGenreNames} from "../../+state/movie-list/movie-list.models";

@Component({
  selector: 'app-movie-list',
  templateUrl: './movie-list.component.html',
  styleUrls: ['./movie-list.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class MovieListComponent {
  @Input() isLoading: boolean;
  @Input() errorMessage: string;
  @Input() movieList: MovieListEntity[];
  @Input() selectedMovie: MovieListEntityWithGenreNames;

  @Input() page: number;
  @Input() pageSize: number;
  @Input() paginationCollectionSize: number;

  @Output() onPageChange = new EventEmitter<number>();
  @Output() onMovieClick = new EventEmitter<number>();
  @Output() onReloadList = new EventEmitter<number>();

  constructor() {
  }

  public trackByFnMovieList(index, item) {
    return item.id;
  }
}
