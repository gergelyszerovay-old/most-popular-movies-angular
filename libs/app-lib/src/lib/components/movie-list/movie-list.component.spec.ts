import {MovieListComponent} from './movie-list.component';
import {async, ComponentFixture, TestBed} from '@angular/core/testing';

import {NgbPagination, NgbToast} from "@ng-bootstrap/ng-bootstrap";

import * as Stories from './movie-list.component.stories';
import {By} from "@angular/platform-browser";
import {MockComponent, ngMocks} from "ng-mocks";

describe('MovieListComponent', () => {
  let component: MovieListComponent;
  let fixture: ComponentFixture<MovieListComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        MovieListComponent,
        MockComponent(NgbPagination),
        MockComponent(NgbToast)
      ]
    })
      .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MovieListComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('trackByFnMovieList should return item.id', () => {
    expect(component.trackByFnMovieList(1, {id: 2})).toBe(2);
  });

  describe('@Output tests', () => {
    it('onMovieClick', () => {
      const props = Stories['isLoaded']().props;
      const {onMovieClick, ...propsFiltered} = props;
      component = Object.assign(component, propsFiltered);
      fixture.detectChanges();

      let selectedMovieId: number;
      component.onMovieClick.subscribe((id: number) => (selectedMovieId = id));

      const debugEl = fixture.debugElement.query(
        By.css('[data-qa="movie-list-ul"] > li:first-child > a')
      );

      debugEl.triggerEventHandler('click', null);
      expect(selectedMovieId).toEqual(props.movieList[0].id);
    });

    it('onPageChange', () => {
      const props = Stories['isLoaded']().props;
      const {onPageChange, ...propsFiltered} = props;
      component = Object.assign(component, propsFiltered);
      fixture.detectChanges();

      const mockedNgbPagination = ngMocks.find<NgbPagination>(fixture.debugElement, 'ngb-pagination').componentInstance;

      let selectedPageId: number;
      component.onPageChange.subscribe((id: number) => {
        selectedPageId = id
      });

      mockedNgbPagination.pageChange.emit(2)
      expect(selectedPageId).toEqual(2);
    });
  });

  describe('Storybook stories snapshot tests', () => {
    for (const [key, value] of Object.entries(Stories)) {
      if (value instanceof Function) {
        it(`Story ${key}`, () => {
          component = Object.assign(component, Stories[key]().props)
          fixture.detectChanges();
          expect(fixture).toMatchSnapshot();
        });
      }
    }
  });

});
