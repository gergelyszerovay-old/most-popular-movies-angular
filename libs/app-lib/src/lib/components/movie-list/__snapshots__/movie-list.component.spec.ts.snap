// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListComponent Storybook stories snapshot tests Story isError 1`] = `
<app-movie-list
  errorMessage={[Function String]}
  isLoading="false"
  onMovieClick={[Function Function]}
  onPageChange={[Function Function]}
  onReloadList={[Function Function]}
>
  <div
    class="c-movie-list x-load-error"
  >
    <ngb-toast
      class="bg-danger text-light"
    >
       Error: Cannot load the movie list from the server!
      <br />
      <a
        href="#"
      >
        Click here to retry.
      </a>
    </ngb-toast>
  </div>
</app-movie-list>
`;

exports[`MovieListComponent Storybook stories snapshot tests Story isLoaded 1`] = `
<app-movie-list
  errorMessage=""
  isLoading="false"
  movieList={[Function Array]}
  onMovieClick={[Function Function]}
  onPageChange={[Function Function]}
  onReloadList={[Function Function]}
  page={[Function Number]}
  pageSize={[Function Number]}
  paginationCollectionSize={[Function Number]}
>
  <div
    class="c-movie-list x-no-load-error"
  >
    <ul
      data-qa="movie-list-ul"
    >
      
      <li>
        <a
          href="#"
        >
          <span>
            Project Power (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <span>
            Tenet (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <span>
            Sputnik (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
    </ul>
    
    <ngb-pagination />
  </div>
</app-movie-list>
`;

exports[`MovieListComponent Storybook stories snapshot tests Story isLoadedWithSelectedMovie 1`] = `
<app-movie-list
  errorMessage=""
  isLoading="false"
  movieList={[Function Array]}
  onMovieClick={[Function Function]}
  onPageChange={[Function Function]}
  onReloadList={[Function Function]}
  page={[Function Number]}
  pageSize={[Function Number]}
  paginationCollectionSize={[Function Number]}
  selectedMovie={[Function Object]}
>
  <div
    class="c-movie-list x-no-load-error"
  >
    <ul
      data-qa="movie-list-ul"
    >
      
      <li>
        <a
          href="#"
        >
          <span>
            Project Power (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          class="x-selected"
          href="#"
        >
          <span>
            Tenet (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <span>
            Sputnik (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
    </ul>
    
    <ngb-pagination />
  </div>
</app-movie-list>
`;

exports[`MovieListComponent Storybook stories snapshot tests Story isLoading 1`] = `
<app-movie-list
  errorMessage=""
  isLoading={[Function Boolean]}
  movieList={[Function Array]}
  onMovieClick={[Function Function]}
  onPageChange={[Function Function]}
  onReloadList={[Function Function]}
  page={[Function Number]}
  pageSize={[Function Number]}
  paginationCollectionSize={[Function Number]}
>
  <div
    class="c-movie-list x-no-load-error"
  >
    <ul
      class="x-is-loading"
      data-qa="movie-list-ul"
    >
      
      <li>
        <a
          href="#"
        >
          <span>
            Project Power (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <span>
            Tenet (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
      <li>
        <a
          href="#"
        >
          <span>
            Sputnik (2020)
          </span>
          <svg
            class="bi bi-caret-right-fill"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.14 8.753l-5.482 4.796c-.646.566-1.658.106-1.658-.753V3.204a1 1 0 0 1 1.659-.753l5.48 4.796a1 1 0 0 1 0 1.506z"
            />
          </svg>
        </a>
      </li>
    </ul>
    
    <ngb-pagination />
  </div>
</app-movie-list>
`;
